urld829b80034_wjch111_PTSP2013Wang_tgz-pJ8-controllers_momcts_momctsCore_TestPlannerJ8/src/controllers/momcts/utils/Logger.java

```diff
@@ -0,0 +1,58 @@
+package controllers.momcts.utils;
+
+import java.io.FileWriter;
+import java.io.IOException;
+
+/**
+ * This class specializes in writing logs to the indicated files in the program
+ * @author wj
+ * @date	created on 2013/07/18
+ * @date	modified on 2013/07/18
+ *
+ */
+public class Logger {
+	private String logFile;
+	/**
+	 * Defines if the logs should be writen as addition of the indicated file, or as replacement of the file
+	 */
+	private boolean append;
+	private FileWriter fstream;
+	
+	private void initialize(){
+		try {
+			fstream = new FileWriter(logFile, append);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *fstream*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).